internal/ingress/controller/parser/parser.go

```diff
@@ -590,6 +623,70 @@ func overrideRoute(route *Route,
 	}
 }
 
+// normalizeProtocols prevents users from mismatching grpc/http
+func normalizeProtocols(route *Route) {
+	protocols := route.Protocols
+	var http, grpc bool
+
+	for _, protocol := range protocols {
+		if strings.Contains(*protocol, "grpc") {
+			grpc = true
+		}
+		if strings.Contains(*protocol, "http") {
+			http = true
+		}
+		if sanitizeProtocol(*protocol) != true {
+			http = true
+		}
+	}
+
+	if grpc && http {
+		route.Protocols = kong.StringSlice("http", "https")
+	}
+	// length check
+	if len(route.Protocols) < 1 {
+		return
+	}
```

Comment:
What is the purpose of this `if` block?
There will be no change in the behavior if the block is present or not.